std/experimental/color/xyz.d

```diff
@@ -0,0 +1,254 @@
+    }
+
+    // comparison
+    bool opEquals(C)(C rh) if(isColor!C)
+    {
+        auto c = to!(typeof(this))(rh);
+        return tristimulus == c.tristimulus;
+    }
+
+    // operators
+    typeof(this) opBinary(string op, S)(S rh) const if(isFloatingPoint!S && (op == "*" || op == "/" || op == "^^"))
+    {
+        alias T = Unqual!(typeof(this));
+        T res = this;
+        foreach(c; xyYComponents)
+            mixin(ComponentExpression!("res._ #= rh;", c, op));
+        return res;
+    }
+    ref typeof(this) opOpAssign(string op, S)(S rh) if(isFloatingPoint!S && (op == "*" || op == "/" || op == "^^"))
+    {
+        foreach(c; xyYComponents)
+            mixin(ComponentExpression!("_ #= rh;", c, op));
+        return this;
+    }
+}
+
+unittest
+{
+    alias XYZf = XYZ!float;
+    alias xyYf = xyY!float;
+
+    // TODO: test xxY operators and functions..
+    static assert(xyYf(0.5, 0.5, 1) * 100.0 == xyYf(50, 50, 100));
+    static assert(xyYf(XYZf(0.5, 1, 0.5)) == xyYf(0.25, 0.5, 1));
+}
+
+
+// std.conv style to!XYZ function
+To to(To, From)(From color) if(isXYZ!To && isXYZ!From)
+{
+    alias F = To.ComponentType;
+    return To(F(color.X), F(color.Y), F(color.Z));
+}
+To to(To, From)(From color) if(isxyY!To && isxyY!From)
+{
+    alias F = To.ComponentType;
+    return To(F(color.x), F(color.y), F(color.Y));
+}
+
+To to(To, From)(From color) if(isXYZ!To && isxyY!From)
+{
+    alias F = To.ComponentType;
+    return To(F(color.Y/color.y*color.x), F(color.Y), F(color.Y/color.y*(F(1)-color.x-color.y)));
+}
+To to(To, From)(From color) if(isXYZ!To && isRGB!From)
+{
+    import std.experimental.color.rgb: RGBColorSpaceDefs;
+
+    alias ToType = To.ComponentType;
+    alias FromType = From.ComponentType;
+    alias WorkType = WorkingType!(FromType, ToType);
+
+    // unpack the working values
+    auto src = color.tristimulus;
+    WorkType r = convertPixelType!WorkType(src[0]);
+    WorkType g = convertPixelType!WorkType(src[1]);
+    WorkType b = convertPixelType!WorkType(src[2]);
+
+    static if(From.linear == false)
+    {
+        r = toLinear!(From.colorSpace)(r);
+        g = toLinear!(From.colorSpace)(g);
+        b = toLinear!(From.colorSpace)(b);
+    }
+
+    // transform to XYZ
+    enum toXYZ = RGBColorSpaceMatrix!(From.colorSpace, WorkType);
+    WorkType[3] v = multiply(toXYZ, [r, g, b]);
+    return To(v[0], v[1], v[2]);
+}
+
+To to(To, From)(From color) if(isxyY!To && isXYZ!From)
+{
+    alias F = To.ComponentType;
+    auto sum = color.X + color.Y + color.Z;
+    return To(F(color.X/sum), F(color.Y/sum), F(color.Y));
+}
+To to(To, From)(From color) if(isxyY!To && isRGB!From)
+{
+    return to!To(XYZ!(To.ComponentType)(color));
+}
+
+
+private:
+
+// shorthand alias for all components
+alias XYZComponents = TypeTuple!('X','Y','Z');
+alias xyYComponents = TypeTuple!('x','y','Y');
+
+// build mixin code to perform expresions per-element
+template ComponentExpression(string expression, char component, string op)
```

Comment:
Yeah yeah. I need somewhere common to put this stuff... `package.d` doesn't feel right.